tests/tasks/setup/kitctl/dataplane.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: data-plane-setup
+  namespace: tekton-pipelines
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/categories: Kubernetes
+    tekton.dev/tags: CLI, kubectl
+    tekton.dev/displayName: "kubernetes actions"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: |
+    Setup a kubernetes data plane in the guest cluster.
+  params:
+  - name: name
+    default: "guest"
+    description: Name of the guest cluster
+  - name: substrate-name
+    description: Name of the substrate cluster
+  - name: node-count
+    default: "5"
+    description: Number of worker nodes
+  steps:
+  - name: setup-data-plane
+    image: bitnami/kubectl
+    script: |
+      #!/bin/bash
+      echo "Setting up data plane"
+      cat <<EOF | kubectl apply -f -
+      apiVersion: kit.k8s.sh/v1alpha1
+      kind: DataPlane
+      metadata:
+        name: $(params.name)
+        namespace: $(params.name)
+      spec:
+        clusterName: $(params.name) # Associated Cluster name
+        nodeCount: $(params.node-count)
+        subnetSelector:
+          karpenter.sh/discovery: $(params.substrate-name)
```

Comment:
It breaks encapsulation to pass the `substrate-name` concept through. Can we call this `discovery-name` or something similar? The value we pass in can be `substrate-name`.

Reply:
I have modified the data plane creation task to auto-detect the subnet and security group selector. Now we no longer need the name of the management cluster.